src/brsTypes/components/RoSGNode.ts

```diff
@@ -1134,8 +1142,12 @@ export function createNodeByType(interpreter: Interpreter, type: BrsString): RoS
         while (typeDef) {
             let init: BrsType;
 
-            addChildren(interpreter, node, typeDef);
-            addFields(interpreter, node, typeDef);
+            interpreter.inSubEnv(subInterpreter => {
+                addChildren(subInterpreter, node!, typeDef!);
```

Comment:
Typescript did not seem to believe that `node` and `typeDef` were defined once I put them in this anonymous function, so I had to add the `!`.

Reply:
Straaaaange